pkg/manager/manager_lockfile.go

```
@@ -0,0 +1,205 @@
+		return fmt.Errorf("failed to serialize lockfile: %w", err)
+	}
+
+	err = os.WriteFile(lockPath, data, 0o644)
+	if err != nil {
+		return fmt.Errorf("failed to write lockfile: %w", err)
+	}
+	return nil
+}
+
+// SetLockFileLocation changes the lockfile location setting and ensures the lockfile exists.
+func SetLockFileLocation(cursorDir string, useRoot bool) (string, error) {
```

golangci-lint (1.64.8)

cognitive complexity 21 of func `SetLockFileLocation` is high (> 20)

(gocognit)